src/screens/MainApp/MyDiagnoses/index.js

```diff
@@ -1,74 +1,69 @@
 import React, { useState, useEffect, useRef } from 'react'
 import { View, FlatList, ActivityIndicator } from 'react-native'
 import { createStackNavigator } from 'react-navigation-stack'
-import { isNull, isEmpty } from 'lodash'
+import { isEmpty } from 'lodash'
 import * as AnalyticsService from '~/analyticsService'
-import { ForceRerenderOnNavigation } from '~/navigationService'
 import * as DatabaseService from '~/databaseService'
-import { sortDiagnosesByMostRecentCreation } from '~/mixins/diagnose'
-import { getURL } from '~/mixins/storage'
-import { renderDiagnoses } from './renderUtilities'
+import { renderDiagnose } from './renderUtilities'
 import Background from '~/components/Background'
 import NoDiagnoses from './components/NoDiagnoses'
 import Header from './components/Header'
 import DetailedDiagnose from '../DetailedDiagnose'
 import { OFFSET_THRESHOLD_TO_SNAP_FLATLIST } from './constants'
 import styles from './styles'
 
+const buildDiagnose = (doc) => {
+  const diagnose = doc.data({ serverTimestamps: 'estimate' })
+  diagnose.thumbnail = diagnose.imageReferences[0]
+  return diagnose
+}
+
 const MyDiagnoses = () => {
   const flatListRef = useRef()
-  const [diagnoses, setDiagnoses] = useState(null)
+  const diagnoses = useRef([])
   const [downloadingDiagnoses, setDownloadingDiagnoses] = useState(true)
-  const [downloadingDiagnosesWhenNoDiagnoses, setDownloadingDiagnosesWhenNoDiagnoses] = useState(true)
-  const [refetch, toggleRefetch] = useState(null)
-  const [refreshing, setRefreshing] = useState(false)
   const [scrolling, setScrolling] = useState(false)
 
   AnalyticsService.setCurrentScreenName('My Diagnoses')
 
-  const refetchDiagnoses = () => {
-    toggleRefetch(!refetch)
-  }
+  useEffect(
+    () => {
+      const unsuscribe = DatabaseService.fetchDiagnosesFromCurrentUser(async (snapshot) => {
+        setDownloadingDiagnoses(true)
+        snapshot.docChanges().map(docChange => {
+          const doc = docChange.doc
+          const changeType = docChange.type
+          const newDiagnoses = diagnoses.current
 
-  const getRenderedDiagnoses = async () => {
-    setDownloadingDiagnosesWhenNoDiagnoses(true)
-    setRefreshing(true)
-    try {
-      const downloadedData = await DatabaseService.getDiagnosesFromCurrentUser()
-
-      const downloadedDataWithThumbnail = await Promise.all(
-        downloadedData.map(async (diagnose, index) => {
-          diagnose.thumbnail = await getURL(diagnose.imageReferences[0])
-          return diagnose
-        })
-      )
+          if (changeType === 'added') {
```

Comment:
What do you think of replacing this `if...else...if...else` chain with a `switch` statement and moving the logic for each case to its own function?
```javascript
switch (changeType) {
  case 'added': setDiagnoses(diagnoses => processNewDiagnose(previousDiagnoses, diagnose)); break
  case 'modified': setDiagnoses(diagnoses => processModifiedDiagnose(previousDiagnoses, diagnose)); break
  case 'removed': setDiagnoses(diagnoses => processRemovedDiagnose(previousDiagnoses, diagnose)); break
  default: throw new Error('Error processing snapshot from Firestore')
}

// All the processXXXDiagnose functions return a new array with the diagnoses
```

Reply:
Definitly it's cleaner

Reply:
default clause it's not needed